netket/utils/jax.py

```diff
@@ -20,3 +20,15 @@ def get_afun_if_module(mod_or_fun, *args, **kwargs):
         return mod_or_fun.apply
     else:
         return mod_or_fun
+
+
+class WrappedApplyFun:
+    def __init__(self, module):
+        self.apply = module
```

Comment:
You should define equality and hashing here otherwise jax will recompile tons of stuff.

Reply:
define them based on equality and hashing of the module